src/pages/index.js

```diff
@@ -103,22 +147,92 @@ function handleOverlay(event) {
 }
 
 function handleEditFormSubmit(evt) {
+    const editFormSubmitBtn = evt.submitter;
+    setButtonText(editFormSubmitBtn, true);
+
     evt.preventDefault();
-    profileName.textContent = editProfileName.value;
-    profileDescription.textContent = editProfileDescription.value;
-    closeModal(editProfileModal);
+    api.editUserInfo({ name: editProfileName.value, about: editProfileDescription.value })
+        .then((data) => {
+            profileName.textContent = data.name;
+            profileDescription.textContent = data.about;
+            closeModal(editProfileModal);
+        })
+        .catch(console.error)
+        .finally(() => {
+            setButtonText(editFormSubmitBtn, false);
+        });
 }
 
+function renderCard(item) {
+    const cardEl = getCardElement(item);
+    cardsList.prepend(cardEl);
+}
+
+
 function handleAddCardSubmit(evt) {
+    const addCardSubmitBtn = evt.submitter;
+    setButtonText(addCardSubmitBtn, true);
+
     evt.preventDefault();
-    const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-    const cardEl = getCardElement(inputValues);
-    cardsList.prepend(cardEl);
-    evt.target.reset();
-    disableButton(cardSubmitButton, settings);
-    closeModal(addCardModal);
+    api.addCard({ name: cardNameInput.value, link: cardLinkInput.value })
+        .then((cardData) => {
+            renderCard(cardData);
+            disableButton(cardSubmitButton, settings);
+            closeModal(addCardModal);
+        })
+        .catch(console.error)
+        .finally(() => {
+            setButtonText(addCardSubmitBtn, false);
+        });
+
 }
 
+function handleAvatarSubmit(evt) {
+    const avatarSubmitBtn = evt.submitter;
+    setButtonText(avatarSubmitBtn, true);
+
+    evt.preventDefault();
+    api.editAvatarInfo(avatarInput.value)
+        .then((avatarData) => {
+            if (avatarData.avatar) {
+                avaterImg.src = avatarData.avatar;
+                console.log(avatarData.avatar);
+                avaterImg.alt = "updated avatar";
```

Comment:
`avaterImg.alt = "updated avatar";
`

Every `img` tag must have a **descriptive** `alt` attribute (the **name** of the image is usually there) to show the **alternative** text if the image is not uploaded, and it’s needed for **screen readers**

```js

avaterImg.alt = profileName.textContent;

```